### PR TITLE
os/bluestore: print debug iterater only need.

### PR DESCRIPTION
**Pull request by majianpeng**
Signed-off-by: Jianpeng Ma <jianpeng.ma@intel.com>

### CODE REVIEW

**Comment by majianpeng** [comment]
@tchaikov . I'm not sure this whether GCC can optimize. Could you review this? Thanks!

File: src/os/bluestore/KernelDevice.cc [hunk 1]
```diff
@@ -645,9 +645,11 @@ int KernelDevice::aio_write(
       ++injecting_crash;
     } else {
       bl.prepare_iov(&aio.iov);
-      for (unsigned i=0; i<aio.iov.size(); ++i) {
-	dout(30) << "aio " << i << " " << aio.iov[i].iov_base
-		 << " " << aio.iov[i].iov_len << dendl;
+      if (cct->_conf->subsys.should_gather(ceph_subsys_bdev, 30)) {
```
Comment (ifed01) [inline src/os/bluestore/KernelDevice.cc]: nit: IMO a bit better to define
const int LEVEL = 30;
and use it for both if and dout..

**Comment by majianpeng** [comment]
@ifed01, update by your suggestion.

File: src/os/bluestore/KernelDevice.cc [hunk 2]
```diff
@@ -712,9 +716,11 @@ int KernelDevice::aio_read(
     ++ioc->num_pending;
     aio_t& aio = ioc->pending_aios.back();
     aio.pread(off, len);
-    for (unsigned i=0; i<aio.iov.size(); ++i) {
-      dout(30) << "aio " << i << " " << aio.iov[i].iov_base
-	       << " " << aio.iov[i].iov_len << dendl;
+    if (cct->_conf->subsys.should_gather(ceph_subsys_bdev, DEBUG_LEVEL)) {
```
Comment (tchaikov) [inline src/os/bluestore/KernelDevice.cc]: @majianpeng @ifed01 probably #18873 is better alternative to address this issue?

**Comment by liewegas** [comment]
closed in favor of #18873

**Pull request closed without merging**